tests/testthat/test-newton_solver.R

```diff
@@ -37,66 +61,19 @@ test_that("Final size with newton solver", {
   )
   # check that solver returns values within range
   expect_true(
-    all(epi_outcome$p_infected > 0)
+    all(epi_outcome$p_infected >= 0.0)
   )
   expect_true(
-    all(epi_outcome$p_infected < 1)
-  )
-  # check for size of the vector
-  expect_length(
-    epi_outcome$p_infected,
-    length(demography_vector) * ncol(psusc)
+    all(epi_outcome$p_infected <= 1.0)
   )
 })
 
-# basic test that newton solver returns correct answer
-test_that("Newton solver returns correct answer", {
-  # prepare some data for the solver
-  r0 <- 1.3
-
-  epi_outcome <- final_size(
-    r0 = r0,
-    contact_matrix = contact_matrix,
-    demography_vector = demography_vector,
-    susceptibility = susc,
-    p_susceptibility = psusc,
-    solver = "newton"
-  )
-
-  epi_outcome_known <- 1 - exp(-r0 * epi_outcome$p_infected)
-
-  tolerance <- 1e-5
-  expect_equal(
-    epi_outcome$p_infected, epi_outcome_known,
-    tolerance = tolerance
-  )
-})
-
-# basic test that higher r0 results in higher final size
-test_that("Newton solver gives higher final size for large r0", {
-  # prepare some data for the solver
-  r0_low <- 1.3
-  r0_high <- 3.3
-
-  epi_outcome_low <- final_size(
-    r0 = r0_low,
-    contact_matrix = contact_matrix,
-    demography_vector = demography_vector,
-    susceptibility = susc,
-    p_susceptibility = psusc,
-    solver = "newton"
-  )
-
-  epi_outcome_high <- final_size(
-    r0 = r0_high,
-    contact_matrix = contact_matrix,
-    demography_vector = demography_vector,
-    susceptibility = susc,
-    p_susceptibility = psusc,
-    solver = "newton"
-  )
-
+### Test that lower susceptibility leads to lower final size ####
+# this is a test of statistical correctness but placed here for convenience
+test_that("Lower susceptibility leads to lower final size", {
+  # all fully susceptibles must have larger final size than immunised
   expect_true(
-    all(epi_outcome_high$p_infected > epi_outcome_low$p_infected)
+    all(epi_outcome[epi_outcome$susc_grp == "susceptible", ]$p_infected >
+      epi_outcome[epi_outcome$susc_grp == "immunised", ]$p_infected)
```

Comment:
Shouldn't this be part of `test-statistical_correctness.R`?

Reply:
Yes it is is, but rather than replicate this test there, I have written it here with a comment on line 72.

Reply:
Now removed, and added a similar test to `test-statistical_correctness.R`.